exporter/opensearchexporter/README.md

```diff
@@ -0,0 +1,66 @@
+# OpenSearch Exporter
+
+| Status                   |           |
+| ------------------------ |-----------|
+| Stability                | [devel]   |
+| Supported pipeline types | traces    |
+| Distributions            | [contrib] |
+
+This exporter supports sending OpenTelemetry signals as documents to [OpenSearch](https://www.opensearch.org).
+
+The documents are sent using [observability catalog](https://github.com/opensearch-project/opensearch-catalog/tree/main/schema/observability) schema.
+
+## Configuration options
+
+- `endpoints`: List of OpenSearch URLs. If endpoints is missing, the
+  OPENSEARCH_URL environment variable will be used.
+- `num_workers` (optional): Number of workers publishing bulk requests concurrently.
+- `dataset` (default=`default`) a user-provided label. It is used to construct the name of the destination index or data stream.
+- `namespace` (default=`namespace`)a user-provided label. It is used to construct the name of the destination index or data stream.
+- `flush`: Event bulk buffer flush settings
+    - `bytes` (default=5242880): Write buffer flush limit.
+    - `interval` (default=30s): Write buffer time limit.
+- `retry`: Event retry settings
+    - `enabled` (default=true): Enable/Disable event retry on error. Retry
+      support is enabled by default.
+    - `max_requests` (default=3): Number of HTTP request retries.
+    - `initial_interval` (default=100ms): Initial waiting time if a HTTP request failed.
+    - `max_interval` (default=1m): Max waiting time if a HTTP request failed.
+
+### HTTP settings
+
+- `read_buffer_size` (default=0): Read buffer size.
+- `write_buffer_size` (default=0): Write buffer size used when.
+- `timeout` (default=90s): HTTP request time limit.
+- `headers` (optional): Headers to be send with each HTTP request.
+
+### Security and Authentication settings
```

Comment:
Rather than directly implementing HTTP basic auth it would be better to use the [`configauth` package](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configauth/README.md) (which is [included in](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L56) the `confighttp.HTTPClientSettings` mentioned above) which can be used to provide a [`basicauthextension`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/basicauthextension) reference or any other client authentication extension.  This will provide significantly more flexibility to users.

Reply:
These settings are passed to OpenSearch Go client. I'll check if it's possible to use `configauth` here instead.

Reply:
Ideally the exporter would be able to provide an `http.Client` to the OpenSearch Go client.  If that's not possible at this time would it be possible to expose such affordances in the OpenSearch Go client library?

Reply:
At the moment, opensearch-go manages HTTP client internally.

Exporter could use settings from `configauth` and initialize opensearch client from those. Would that make sense?Refactoring opensearch-go client is a larger project.